lib/cylc/scheduler_cli.py

```diff
@@ -162,6 +162,17 @@ def parse_commandline(is_restart):
         help="Specify the suite source.",
         metavar="SOURCE", action="store", dest="source")
 
+    parser.add_option(
+        "--host-select",
+        help="Enable host selection functionality.",
+        action="store_true", default=False, dest="host_select")
```

Comment:
@cylc/core How do we want to go about this?

* I guess host selection should be True by default.
* We should be able to overwrite this with `--host` as required. This would make `--no-host-select` similar to `--host=localhost`.
* Should we exit 1 if both are specified.

Reply:
(discussed in person)

Reply:
**Suggestion: Always use host selection unless overridden by the `--host` setting.**

The default list of suite servers is `["localhost"]` so this is no change from current behaviour if the default is unchanged.

```bash
# no servers configured in user/site config - use localhost
cylc run <suite>

# servers configured - use host selection
cylc run <suite>

# override host selection by manually specifying a host
cylc run --host=localhost
```

We would no longer require the `--host-select` and `--no-host-select` settings.

This should be fine unless there is any reason why we might want to invoke host selection on the host specified by `--host`? I guess such requirements would probably be mitigated by future architecture anyway?

Reply:
The approach I've taken with the health check is to introduce a "condemned hosts" configuration so that hosts have to be explicitly stated before suites jump ship. I think this removes the requirement to disable the health check. Is there a case where we would want to switch off the server health check?

``--no-host-select`` inferring no health check is somewhat implicit, if required we might want to consider something like ``--no-health-check``?

Reply:
Initially I implemented the options according to Matt's suggestions, but if this is up for debate, here are my thoughts. It seems to me, in agreement with @oliver-sanders, that the cases are as follows:

#### Launching a suite
1. run on specified ``--host=<HOST>`` (including ``HOST=localhost``)
2. with no specification as in [2]: look at the global config:
* with default config, run on localhost (hence it remains the default host choice overall)
* with multiple hosts specified, use host selection to choose one from those specified. (**Q:** if host selection is disabled & ``--host`` is not provided, how would we choose a host from non-default multi-host config? The only option I see is that localhost is therefore implied.)

#### Host health check / reconsideration*
*probably over-simplified, please correct me if necessary.
1. Look at the health of the hosts in order to reconsider & possibly change run host.
2. Not do any health-checking, remaining on the same host as launched on.

Therefore, could we not limit the options to ``--host``, for the usual specification, enabling future health checking, and ``--fixed-host`` (or similar), launching on a permanent host i.e. with health checking disabled? Otherwise host selection would be enabled (& necessary) by default.

This is essentially that is as suggested by Oliver, but with the ``--no-host-select``/``--no-health-check`` option given what I think is a more intuitive name.

Reply:
(discussed & decided in person)